FWCore/Framework/src/ExceptionActions.cc

```diff
@@ -8,23 +8,16 @@
 
 namespace edm {
   namespace exception_actions {
-    namespace {
-      struct ActionNames {
-        ActionNames() : table_(LastCode + 1) {
-          table_[IgnoreCompletely] = "IgnoreCompletely";
-          table_[Rethrow] = "Rethrow";
-          table_[SkipEvent] = "SkipEvent";
-          table_[FailPath] = "FailPath";
-        }
-
-        typedef std::vector<char const*> Table;
-        Table table_;
-      };
-    }  // namespace
-
     char const* actionName(ActionCodes code) {
-      static ActionNames tab;
-      return static_cast<unsigned int>(code) < tab.table_.size() ? tab.table_[code] : "UnknownAction";
+      static const std::array<char const*, LastCode> tab = []() {
+        std::array<char const*, LastCode> table;
+        table[IgnoreCompletely] = "IgnoreCompletely";
+        table[Rethrow] = "Rethrow";
+        table[SkipEvent] = "SkipEvent";
+        table[FailPath] = "FailPath";
+        return table;
+      }();
```

Comment:
@Dr15Jones What do you think about this change?

Reply:
Could it be constexpr?

Reply:
Done. I want to note that I was tempted to use the initializer list to initialize the array, but I valued more the consistency of taking the indices of elements from the enum.